src/routing/approved.rs

```diff
@@ -1028,39 +1023,64 @@ impl Approved {
                 (MIN_AGE + 1, None, None)
             };
 
-        if let Some(ResourceProofResponse {
-            solution,
-            data,
-            nonce,
-            nonce_signature,
-        }) = join_request.resource_proof_response
-        {
-            let serialized = bincode::serialize(&(*peer.name(), nonce))?;
-            if self
-                .node
-                .keypair
-                .public
-                .verify(&serialized, &nonce_signature)
-                .is_ok()
-                && self.resource_proof.validate_all(&nonce, &data, solution)
-            {
-                return self.vote(Vote::Online {
-                    member_info: MemberInfo::joined(peer.with_age(age)),
-                    previous_name,
-                    their_knowledge,
-                });
+        // Require resource proof only if joining as a new node.
+        if previous_name.is_none() {
+            if let Some(response) = join_request.resource_proof_response {
```

Comment:
seems we could use something like:
```rust
if let (true, Some(response)) = (previous_name.is_none(), join_request.resource_proof_response) {}
```
to avoid net if

Reply:
Do you mean something like this:
```rust
if let (true, Some(response)) = (previous_name.is_none(), join_request.resource_proof_response) {
    self.validate_resource_proof_response(...)
} else {
    self.send_resource_proof_challenge(...)
}
```
?
Because that would send the challenge even to relocated nodes which we don't want.
```

Reply:
Ah, messed up with my previous logic.
you are right, they cannot be bound.